examples/perlock-single-simple-demo/src/main/java/name/mitterdorfer/perlock/examples/PathWatcherDemo.java

```diff
@@ -0,0 +1,44 @@
+package name.mitterdorfer.perlock.examples;
+
+import name.mitterdorfer.perlock.PathWatcher;
+import name.mitterdorfer.perlock.SinglePathWatcherFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * This example application demonstrates how to create and use a path watcher. The demo accepts two command line
+ * parameters:
+ *
+ * <ol>
+ *     <li><code>--recursive</code> to watch a path recursively or <code>--non-recursive</code> to watch
+ * non-recursively (i.e. only in the provided path but not below)</li>
+ *      <li><code>path</code> which is the path to watch</li>
+ * </ol>
+ *
+ * If you invoke the sample application as follows, it watches the home folder (on Unix): <code>java -jar perlock-examples.jar --non-recursive ~</code>
+ */
+class SinglePathWatcherDemo {
```

Comment:
I think the class name should match the file name. Is it intentionally that the class is not declared `public`?